### PR TITLE
Document why Gradient.reverse() doesn't reverse constant gradients

### LINKED ISSUE #76169: Gradient.reverse() produces unexpected result on constant gradients
### Godot version

4.0.2

### System information

All systems

### Issue description

https://user-images.githubusercontent.com/85438892/232497823-8a87f39f-eb15-4fca-a1d8-356d3af26247.mp4

This behavior of `Gradient.reverse()` is quite surprising at first. It's caused by the points being mirrored along the gradient's middle, but maybe it should be different for constant interpolation.

### Steps to reproduce

Add a gradient, make its interpolation constant, use the Reverse Gradient button.

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by MewPurPur**
Edit: Closes #76169

Documents this behavior:

https://user-images.githubusercontent.com/85438892/232495416-96b97a62-d203-442b-aee4-1ff9f3f9f6f1.mp4

Or maybe we decide that this a bug that should be fixed. (Edit: I recommend that we don't decide this, see comment)

### CODE REVIEW

**Comment by MewPurPur** [comment]
Fundamentally, any attempt to solve the issue presented here would result in users losing points at offset 1. Also, I can actually not think of any way to solve this in a way that maps well beyond the [0, 1] domain (gradients can have points anywhere from `-INF` to `INF`, but only the [0, 1] domain is shown.) Any solutions I thought up of result in the rightmost point being more likely to be mapped to an offset lower to 0.

In short, let's just merge this and not worry about this again. The "bug" is impossible to fix without screwing over users worse than the "bug" screws with them.

**Comment by MewPurPur** [comment]
I'll keep this PR for now, but I believe it shouldn't be merged. The fix with an out-of-domain handle seems better to me and I'll look into implementing it when I have time.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.0.3.
